test/include/seqan3/test/cereal.hpp

```diff
@@ -0,0 +1,59 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides cereal functionality for tests.
+ * \author Enrico Seiler <enrico.seiler AT fu-berlin.de>
+ */
+
+#pragma once
+#if SEQAN3_WITH_CEREAL
+
+#include <gtest/gtest.h>
+
+#include <fstream>
+
+#include <cereal/archives/binary.hpp>
+#include <cereal/archives/portable_binary.hpp>
+#include <cereal/archives/json.hpp>
+#include <cereal/archives/xml.hpp>
+#include <cereal/types/vector.hpp>
+
+#include <seqan3/test/tmp_filename.hpp>
+
+
+namespace seqan3
+{
+
+namespace test
+{
+
+template <typename in_archive_t, typename out_archive_t, typename TypeParam>
+void do_serialisation(TypeParam const l)
+{
+    // This makes sure the file is also deleted if an exception is thrown in one of the tests below.
+    // Generate unique file name.
+    tmp_filename filename{"cereal_test"};
+    {
+        std::ofstream os{filename.get_path(), std::ios::binary};
+        out_archive_t oarchive{os};
+        oarchive(l);
+    }
+
+    {
+        TypeParam in_l{};
+        std::ifstream is{filename.get_path(), std::ios::binary};
+        in_archive_t iarchive{is};
+        iarchive(in_l);
+        EXPECT_EQ(l, in_l);
+    }
+}
+
+} // namespace seqan3::test
+
+} //namespace seqan3
+#endif // SEQAN3_WITH_CEREAL
```

Comment:
can you define these functions as NOPs if cereal isn't defined? Than we can save the checks in every test and just do it here.

Reply:
Good idea, will do

Reply:
But that would mean testing every archive in general, but this should be fine, right?

Reply:
You mean always testing textarchives and binaryarchives? Yes, that would be desirable even, I think.